code/game/objects/items/weapons/switchtool.dm

```diff
@@ -59,6 +62,9 @@
 		choose_deploy(user)
 
 /obj/item/weapon/switchtool/attackby(var/obj/item/used_item, mob/user)
+	if(istype(used_item, /obj/item/weapon/screwdriver) && no_remove)
```

Comment:
This should be `removing_item`, not specifically a screwdriver. Or, better yet, do what I said in the above comment.